### PR TITLE
dkgInfo set before echobroadcast attempted to connect isolated nodes

### PR DESCRIPTION
**Pull request by CluEleSsUK**
dkgInfo was being set on the beacon process after the echobroadcast.
This meant that isolated nodes were not actually informed of the DKG,
and would get excluded in future rounds.
The fix was to reorder some of the setting of dkg info (and locking) in
the dkg and reshare so it would be available for broadcast

### CODE REVIEW

**Comment by AnomalRoil** [comment]
This is actually fixing #873, we confirmed it together with @CluEleSsUK 🎉

File: chain/beacon/callbacks_test.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ func TestStoreCallback(t *testing.T) {
 	require.False(t, checkOne(doneCh))
 
 	cb.RemoveCallback(id1)
+	cb.Close()
```
Comment (AnomalRoil) [inline chain/beacon/callbacks_test.go:37]: This was leaking `nCPU` goroutines from that test.

File: core/drand_beacon_control.go [hunk 2]
```diff
@@ -370,36 +370,43 @@ func (bp *BeaconProcess) runDKG(leader bool, group *key.Group, timeout uint32, r
...
-		// phaser will kick off the first phase for every other nodes so
-		// nodes will send their deals
-		bp.log.Infow("", "init_dkg", "START_DKG")
-		go phaser.Start()
+	board := newEchoBroadcast(bp.log, bp.version, beaconID, bp.privGateway.ProtocolClient,
+		bp.priv.Public.Address(), group.Nodes, func(p dkg.Packet) error {
+			return dkg.VerifyPacketSignature(config, p)
+		})
+
+	dkgProto, err := dkg.NewProtocol(config, board, phaser, true)
```
Comment (AnomalRoil) [inline core/drand_beacon_control.go:400]: The reason why this change works is that `dkg.NewProtocol` is starting immediately in Kyber and uses the broadcast channel directly, so we need to make sure the nodes are ready to receive broadcasts, and for that `dkgInfo` must not be `nil`.

**Review by willscott — COMMENTED** [review]
are the test  failures here something we should be looking into?

**Review by nikkolasg — CHANGES_REQUESTED** [review]
I am not quite sure I understand the problem nor if it is a solution we want to go to.
* [Here](https://github.com/drand/drand/blob/master/core/drand_beacon_public.go#L25) you can see that any nodes that have set their bp.dkgInfo would receive and process the messages received by echo broadcast
* [Here](https://github.com/drand/kyber/blob/master/share/dkg/protocol.go#L127) you can see that `dkg.NewProtocol` doesn't do anything unless the phaser starts. The phaser starts for the leader at the beginning and as soon as a node receives their first msg, they go and start the phaser.

> The reason why this change works is that dkg.NewProtocol is starting immediately in Kyber and uses the * 
broadcast channel directly, so we need to make sure the nodes are ready to receive broadcasts, and for that dkgInfo must not be nil.

* So it's not  true that `dkg.NewProtocol` starts immediatly and use the broadcast channel directly. And even if, where is the problem since the broadcast channel is created. why dkgInfo would be nil ? If it is nil it's because the method `runDKG` has never been called and that's something outside of the DKG world. Maybe in tests the time between "the leader sends the group file" and the time "the DKG starts" is too small and some nodes can't do it fast enough on crappy tests machine ?

TLDR: I'm confused and need more explanation for me to approve this PR.

**Comment by CluEleSsUK** [comment]
Actually the moving of stuff around was due to a sigsev here:
```
github.com/drand/drand/core.(*BeaconProcess).BroadcastDKG(0xc000eec1e0, {0x100e9b088, 0xc0009a6fc0}, 0x1014282e0?)
        /Users/patrickmcclurg/repos/drand/core/drand_beacon_public.go:35 +0x220
github.com/drand/drand/core.(*DrandDaemon).BroadcastDKG(0x100b5b61d?, {0x100e9b088, 0xc0009a6fc0}, 0xc0010df880)
        /Users/patrickmcclurg/repos/drand/core/drand_daemon_public.go:17 +0x9c
github.com/drand/drand/protobuf/drand._Protocol_BroadcastDKG_Handler.func1({0x100e9b088, 0xc0009a6fc0}, {0x100e07e60?, 0xc0010df880})
        /Users/patrickmcclurg/repos/drand/protobuf/drand/protocol_grpc.pb.go:267 +0x84
github.com/drand/drand/core.(*DrandDaemon).NodeVersionValidator(0xc00033a2d0, {0x100e9b088, 0xc0009a6fc0}, {0x100e07e60, 0xc0010df880}, 0x100e30860?, 0xc001b49770)
```

I suspect actually that https://github.com/drand/drand/blob/60b20f67c3ba077c73e0cdea1bdd4b0daf414cbe/core/drand_beacon_control.go#L416 might have something to do with it - on master surely the DKG info will get cleared every time!?

**Comment by CluEleSsUK** [comment]
Closing due to wider dkg refactor work happening
